### PR TITLE
Move sp_update and rot_update to MgsCommon

### PR DESCRIPTION
**Pull request by mkeeter**
It would be convenient for https://github.com/oxidecomputer/hubris/pull/1707 if `MgsCommon` could check update state, so this PR moves the `SpUpdate` and `RotUpdate` members into `MgsCommon`.

### CODE REVIEW

File: task/control-plane-agent/src/mgs_psc.rs [hunk 1]
```diff
@@ -26,7 +26,8 @@ use userlib::sys_get_timer;
 
 // How big does our shared update buffer need to be? Has to be able to handle SP
 // update blocks for now, no other updateable components.
-const UPDATE_BUFFER_SIZE: usize = SpUpdate::BLOCK_SIZE;
+const UPDATE_BUFFER_SIZE: usize =
```
Comment (mkeeter) [inline task/control-plane-agent/src/mgs_psc.rs:29]: This is a drive-by fix to make `UPDATE_BUFFER_SIZE` more robust in case per-component block size ever diverges; `mgs_gimlet.rs` was already doing this.

File: task/control-plane-agent/src/mgs_sidecar.rs [hunk 2]
```diff
@@ -42,7 +42,8 @@ userlib::task_slot!(TRANSCEIVERS, transceivers);
 
 // How big does our shared update buffer need to be? Has to be able to handle SP
 // update blocks for now, no other updateable components.
-const UPDATE_BUFFER_SIZE: usize = SpUpdate::BLOCK_SIZE;
+const UPDATE_BUFFER_SIZE: usize =
+    usize_max(SpUpdate::BLOCK_SIZE, RotUpdate::BLOCK_SIZE);
```
Comment (mkeeter) [inline task/control-plane-agent/src/mgs_sidecar.rs:46]: (same here)

**Review by hawkw — APPROVED** [review]

**Review by jgallagher — APPROVED** [review]
